src/misc/unicode.cpp

```diff
@@ -843,31 +958,16 @@ static bool wide_to_dos(const std::vector<uint16_t>& str_in, std::string& str_ou
 	return status;
 }
 
-static void dos_to_wide(const std::string& str_in,
-                        std::vector<uint16_t>& str_out, const uint16_t code_page)
-{
-	// Unicode code points for screen codes from 0x00 to 0x1f
-	// see: https://en.wikipedia.org/wiki/Code_page_437
-	constexpr uint16_t codes[0x20] = {
-	        0x0020, 0x263a, 0x263b, 0x2665, // 00-03
-	        0x2666, 0x2663, 0x2660, 0x2022, // 04-07
-	        0x25d8, 0x25cb, 0x25d9, 0x2642, // 08-0b
-	        0x2640, 0x266a, 0x266b, 0x263c, // 0c-0f
-	        0x25ba, 0x25c4, 0x2195, 0x203c, // 10-13
-	        0x00b6, 0x00a7, 0x25ac, 0x21a8, // 14-17
-	        0x2191, 0x2193, 0x2192, 0x2190, // 18-1b
-	        0x221f, 0x2194, 0x25b2, 0x25bc, // 1c-1f
-	};
-
-	constexpr uint16_t codepoint_7f = 0x2302;
-
+static void dos_to_wide(const std::string& str_in, std::vector<uint16_t>& str_out,
+                        const DosStringType string_type, const uint16_t code_page)
```

Comment:
Let's not use out args if we can avoid it. And it's trivial to avoid it in this case: just return `std::string`.

Reply:
Some functions from this family returns true/false status, which is being used in locale/translation support. I prefer to keep the API consistent for now. Since I am in the process of reworking locale support, I'll keep this in mind.

Reply:
Sounds good.